src/misc/lv_mem.c

```diff
@@ -204,6 +204,39 @@ void * lv_mem_realloc(void * data_p, size_t new_size)
     return new_p;
 }
 
+bool lv_mem_realloc_safe(void ** data, size_t new_size)
+{
+    void * new_p = NULL;
+
+    MEM_TRACE("reallocating %p with %lu size", *data, (unsigned long)new_size);
+
+    if((*data == NULL) && (0 < new_size)) {
+        *data = lv_mem_alloc(new_size);
+        return false;
+    }
+    else if((*data != NULL) && (0 == new_size)) {
+        MEM_TRACE("using zero_mem");
+        lv_mem_free(*data);
+        *data = &zero_mem;
+        return false;
+    }
+
+#if LV_MEM_CUSTOM == 0
+    new_p = lv_tlsf_realloc(tlsf, data, new_size);
+#else
+    new_p = LV_MEM_CUSTOM_REALLOC(data, new_size);
+#endif
+    if(new_p == NULL) {
+        LV_LOG_ERROR("couldn't allocate memory");
+        return false;
+    }
```

Comment:
All these should be handled in `lv_mem_realloc()`

Reply:
Maybe have a look here and please read func desc. Implementation of safe version shall be super super non-complex. 

https://github.com/MaJerle/lwmem/blob/de2917481ddcb02d4789b188f7c8209af6cdcc17/lwmem/src/lwmem/lwmem.c#L894